src/lib/network_pool/rate_limiter.mli

```diff
@@ -0,0 +1,15 @@
+open Core
+open Network_peer
+
+type t
+
+val create : capacity:int * [`Per of Time.Span.t] -> t
+
+val add :
+     t
+  -> Envelope.Sender.t
+  -> now:Time.t
+  -> score:int
+  -> [`Ok | `Capacity_exceeded]
```

Comment:
I would change `Ok` to something like `Within_capacity`, to avoid confusion with an `Or_error` type.